### PR TITLE
Improved the README.md again

### PR DESCRIPTION
**Pull request by quinton-ashley**
Linking to the demo within the repo won't work unless GitHub Pages are set up on the repo. It shows a 404 error right now. @Birch-san if you want, try setting that up! For now I linked to the online demos made by @ErikSom.

I also put the online demos section first and moved the git clone section to the build from source pre-reqs section.

### CODE REVIEW

File: box2d3-wasm/README.md [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+Check out the [pyramid drop demo](https://box2d-v3-threading.netlify.app/?threading=1&pyramidHeight=64). You can change the size of the pyramid by editing the "pyramidHeight" URL parameter. Set it to 101 for a pyramid with 5,050 boxes!
 
-Install [Node.js and npm][].  
-Install [npm package](https://www.npmjs.com/package/box2d3-wasm) like so:
+Check out the [stress test demo](https://phaser-stress-test-box2d3-wasm.netlify.app/?threading=1). How many shapes can your computer simulate at 60fps? box2d3-wasm is the fastest Box2D implementation on the web.
 
-```bash
-npm i --save box2d3-wasm
-```
+## Web Usage
```
Comment (Birch-san) [inline box2d3-wasm/README.md]: what was wrong with having subheadings? not every section here is orthogonal, we're allowed to have an information hierarchy

**Comment by Birch-san** [comment]
> Linking to the demo within the repo won't work

I wanted to show the users the code, so they can write their own code.

setting up github pages is non-trivial because the package needs to be distributed from somewhere (which then begs the question: which version, and under what circumstance can the version change), and need the ability to serve wasm with the correct mime-type, and serve the index.html with the correct headers for running with threading.

the trickier thing is what should the github pages demo's relationship be with the existing code in the `demo` directory. it would be good to avoid having to maintain multiple duplicate demos, but at the same time they are deployed in totally different ways so it's hard to ensure that demo changes work for both clients.

**Comment by Birch-san** [comment]
github pages doesn't support setting COOP/COEP headers, which prevents the use of threading via SharedArrayBuffer.

https://github.com/orgs/community/discussions/13309  
https://stackoverflow.com/questions/67259043/can-coop-coep-headers-be-set-with-meta-tags-http-equiv

I have a cunning plan

**Comment by quinton-ashley** [comment]
> github pages doesn't support setting COOP/COEP headers, which prevents the use of threading via SharedArrayBuffer.

Ah I was afraid of that. I wonder if a workaround is possible.

File: box2d3-wasm/README.md (hunk 1 above)
Comment (quinton-ashley) [reply box2d3-wasm/README.md]: I think when there's only two subsections, generally that's not enough, there should be at least three to justify doing sub-headings. That's just my opinion though. You could make "Usage" an h2 and then web usage and Node usage the h3s but they were just bold text before.

**Comment by quinton-ashley** [comment]
For now could you approve the PR and make any changes you'd like after, then we can at least showcase the current online demos from Erik?

**Comment by quinton-ashley** [comment]
> I wanted to show the users the code, so they can write their own code.

Also in that case we gotta link to the files on GitHub directly cause links within the repo don't work on the npm package page.

**Comment by Birch-san** [comment]
> For now could you approve the PR and make any changes you'd like after, then we can at least showcase the current online demos from Erik?

just got github pages working; does this change matters?  
https://birch-san.github.io/box2d3-wasm/?threads=4

(can also do `threads=auto` to configure based on `navigator.hardwareConcurrency`, but based on what Erin's saying, 4 may be faster than using all).

**Comment by quinton-ashley** [comment]
@Birch-san oh awesome!

**Comment by quinton-ashley** [comment]
wow!! yeah huge difference actually, runs BLAZING fast now!

New problem though, I can't even make a pyramid high enough to max out the CPU haha. The top of the pyramid is not being created.

https://birch-san.github.io/box2d3-wasm/?threads=auto&pyramidHeight=120

File: box2d3-wasm/README.md [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
-git clone --recurse-submodules https://github.com/Birch-san/box2d3-wasm.git
-```
+## Online Demos
 
-## Install from npm
+Check out the [pyramid drop demo](https://box2d-v3-threading.netlify.app/?threading=1&pyramidHeight=64). You can change the size of the pyramid by editing the "pyramidHeight" URL parameter. Set it to 101 for a pyramid with 5,050 boxes!
 
-Install [Node.js and npm][].  
-Install [npm package](https://www.npmjs.com/package/box2d3-wasm) like so:
+Check out the [stress test demo](https://phaser-stress-test-box2d3-wasm.netlify.app/?threading=1). How many shapes can your computer simulate at 60fps? box2d3-wasm is the fastest Box2D implementation on the web.
```
Comment (Birch-san) [inline box2d3-wasm/README.md]: I wanna be more dignified than this. perf should speak for itself, we don't need to hype it up with invitations or claims. we should prefer to state facts (like we have support for SIMD* and threading) without diminishing other people's efforts. if we have benchmarks we can point at those numbers, but I don't want a tone of getting overexcited at our own results.

*note: we haven't really verified whether the SIMD support is working as intended, so best not to oversell this until we are more confident it's integrated correctly. the previous box2d-wasm got I think <1% speedup from SIMD, and there was evidence that only a small amount of the code was getting vectorized. we're likely to be in a much better situation than this now that box2d is designed for SIMD, but worth measuring first.

Comment (quinton-ashley) [reply box2d3-wasm/README.md]: alright for sure

**Comment by quinton-ashley** [comment]
> 4 may be faster than using all

I'm not seeing a difference yet. I think maybe the auto thread count should be set to `Math.max( 4, navigator.hardwareConcurrency / 2)` though because many Apple CPUs at least have less high proficiency cores than total cores. For example the M1 only has 4 high proficiency cores out of 8 total. Also might be good to keep cores open to avoid disruption from background tasks, what do you think?

File: box2d3-wasm/README.md (hunk 1 above)
Comment (quinton-ashley) [reply box2d3-wasm/README.md]: I added the npm section as the third subsection of the "Usage" section.

**Comment by quinton-ashley** [comment]
I edited the PR, let me know if you want any other changes.

**Comment by Birch-san** [comment]
I'll paraphrase what Erin was saying about thread counts:  
the bottleneck is synchronization of split caches across CCDs.  
some CPUs have more than 8 cores sharing a single cache (some Apple and Intel chips), so we could use high thread counts there.  
but a safer lowest-common-denominator setting for the web could be more like 4 threads.

so the concern wasn't really related to efficiency cores vs performance cores, but yeah maybe even if cache is being shared well we might want to limit ourselves to use no more threads than we have performance cores.

**Comment by quinton-ashley** [comment]
ah nice! so it sounds like 8 threads would be ideal for the M1 then

**Pull request merged**